jcl/src/java.base/share/classes/com/ibm/tools/attach/target/Attachment.java

```diff
@@ -72,15 +72,31 @@
 		static Method startRemoteManagementAgentMethod = null;
 		static {
 			AccessController.doPrivileged((PrivilegedAction<Object>) () -> {
-				Class<?> agentClass;
-				Class<?> startRemoteArgumentType;
+				String agentClassName = 
+						/*[IF Sidecar19-SE]*/
+						"jdk.internal.agent.Agent"; //$NON-NLS-1$
+						/*[ELSE] Sidecar19-SE
+						"sun.management.Agent"; //$NON-NLS-1$
+						/*[ENDIF] Sidecar19-SE */
+				IPC.logMessage("Loading " + agentClassName); //$NON-NLS-1$
 				try {
+					Class<?> agentClass = null;
+					Class<?> startRemoteArgumentType = null;
 					/*[IF Sidecar19-SE]*/
-					agentClass = Class.forName("jdk.internal.agent.Agent"); //$NON-NLS-1$
+					String jmaName = "jdk.management.agent"; //$NON-NLS-1$
+					java.lang.Module jmaModule = jdk.internal.module.Modules.loadModule(jmaName);
+					if (null == jmaModule) {
+						throw new ClassNotFoundException("Cannot load " + jmaName); //$NON-NLS-1$
```

Comment:
Is this really the correct behavior here? What effect will this have on Attach API for subsequent operations?

Reply:
Also it is weird to throw a ClassNotFoundException to describe failing to load a module.

Reply:
> What effect will this have on Attach API for subsequent operations?

This has the same effect as before: attach API per se and other functions like loading agents and getting properties will work, though we won't be able to start the management agents.

> Also it is weird to throw a ClassNotFoundException to describe failing to load a module.

I throw a CNFE because I am encountering an error while trying to find a class.  I am open to suggestions for a more appropriate exception.

Reply:
What does the RI do?

Reply:
`Caused by: com.sun.tools.attach.AttachOperationFailedException: java.lang.module.FindException: Module jdk.management.agent not found`

Reply:
OpenJ9 1st attempt:
`Caused by: com.sun.tools.attach.AttachOperationFailedException: AttachOperationFailedException from target: ATTACH_ERR AttachOperationFailedException in startLocalManagementAgent:  startLocalManagementAgent error starting agent:class java.lang.ExceptionInInitializerError null`

2nd attempt (to same runtime), note the NoClassDefFoundError:
`Caused by: com.sun.tools.attach.AttachOperationFailedException: AttachOperationFailedException from target: ATTACH_ERR AttachOperationFailedException in startLocalManagementAgent:  startLocalManagementAgent error starting agent:class java.lang.NoClassDefFoundError com.ibm.tools.attach.target.Attachment$MethodRefsHolder (initialization failure`

Reply:
I captured exceptions from the module loading code and cleaned up the error handling. The attacher now prints (consistently):
```
jjava.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at Test1a.startAgent(Test1a.java:30)
	at Test1a.main(Test1a.java:17)
Caused by: com.sun.tools.attach.AttachOperationFailedException: AttachOperationFailedException from target: ATTACH_ERR: AttachOperationFailedException in startLocalManagementAgent:
"com.ibm.tools.attach.target.IbmAttachOperationFailedException: Target VM cannot access startLocalManagementAgent"; Caused by: "java.lang.module.FindException: Module jdk.management.agent not found"
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.parseResponse(OpenJ9VirtualMachine.java:347)
	at jdk.attach/com.ibm.tools.attach.attacher.OpenJ9VirtualMachine.startLocalManagementAgent(OpenJ9VirtualMachine.java:537)
	... 6 more
```

Reply:
This message is still very verbose, containing redundant information, and I think it should be cleaned up, but it is a preexisting problem and I'd prefer any further cleanup which is unrelated to this fix to occur in a separate PR.